src/west/_bootstrap/main.py

```diff
@@ -195,6 +196,14 @@ def init_bootstrap(directory, args):
     clone(args.manifest_url, args.manifest_rev,
           os.path.join(directory, WEST_DIR, MANIFEST))
 
+    # Create an initial configuration file
+
+    config = configparser.ConfigParser()
+    config['west'] = {'revision': args.west_rev}
```

Comment:
Why are you storing only `revision` but not `url` (or `remote`)?

Reply:
Not using them anywhere yet. Will add them once a need comes up. Maybe we can just get the information from Git.

Reply:
OK so there's something I don't understand here. Why make `revision` configurable then but not `url`? is it because you can change the revision of a manifest/west installation withou the need for a "reinit"?

Reply:
Because none of the git commands need to look at the URL yet. It'd be dead code.

If they ever do, then we add it then. :)

Reply:
Maybe the reinit thing would use it. To be honest, I'm not even sure how it's supposed to work.

Are people expected to regularly rerun the bootstrapper to do stuff?

Reply:
OK, but if you implement the parsing of the config as suggested, then it would make sense to add the url so that it takes over if the user doesn't specify it via cmd-line options?

Reply:
>Are people expected to regularly rerun the bootstrapper to do stuff?

I think only `reinit` if we choose to implement that as per the current proposal.

Reply:
I believe that if we implement the feature according to the comments https://github.com/zephyrproject-rtos/west/pull/75#discussion_r225202907 and https://github.com/zephyrproject-rtos/west/pull/75#discussion_r225202122, then you would use:

* `revision` always, as you are already doing now (plus of course parsing the conf files)
* `url`/`remote`: when running `west init` only for now (until we decide what to do with `reinit`).

Reply:
Didn't realize we had a `reinit` too. Thought it was rolled into `init`. Same thing goes for that one though.

Configuration-handling-wise, what Git does, and what I think makes the most sense, is to just not write configuration options that aren't inherently repository-specific into `.git/config`. That way, you automatically get them from any global configuration files instead, and from defaults if they don't appear in any configuration files.

With that scheme, there's no need to read the parent configuration files if you're just updating the repository settings. It's pretty neat.

For example, you can make `author` a repository setting if you really want to by adding it to `git/config` (skipping the `--global` flag to `git config`), but `git clone` won't write it to `git/config` (or look at it at all).

So, if we want to make the revision a global setting unless specified (which I still think is a bit obscure and weird though), the neat Git-like way would be to just not write the revision into `west/config` unless explicitly specified.